OADP-8331: Add OADP 1.6.1 RNs (#115703)
openshift/openshift-docs: modules/oadp-1-6-1-release-notes.adoc

Reviewer: Hi Valentina, if time permits, please include a link to the RHSA advisory rather than listing each individual CVE. This will ensure that the RHSA advisory is the source of truth. Also see the email messages about this.
Linking to the RHSA advisory is what OCP and other products do. We want to be consistent with OCP in our Release Notes

Author: Thanks! Correct, we already have a placeholder for the advisory.

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/release-notes/oadp-1-6-release-notes.adoc
:_mod-docs-content-type: REFERENCE

[id="oadp-1-6-1-release-notes_{context}"]
= {oadp-short} 1.6.1 release notes

[role="_abstract"]
{oadp-first} 1.6.1 release notes list resolved and known issues.

The following Red Hat Security Advisory (RHSA) is available for {oadp-short} 1.6.1:

* link:TBD[RHSA-2026:xxxx - New version of Red{nbsp}Hat OpenShift API for Data Protection]

[id="resolved-issues-1-6-1_{context}"]
== Resolved issues

{oadp-short} CLI no longer exposes the namespace flag for non-admin operations::

> Before this update, the {oadp-short} command-line interface (CLI) exposed the global `--namespace` or `-n` flag in the help output of all `oc oadp nonadmin` commands, even though non-admin operations do not support namespace selection due to multi-tenancy boundaries. As a consequence, if you specified the namespace flag during a non-admin backup or restore operation, the CLI ignored the provided value and used the active `oc project` context instead, potentially creating resources in an unintended namespace.

Reviewer 2: ```suggestion
Before this update, the {oadp-short} command-line interface (CLI) exposed the global `--namespace` or `-n` flag in the help output of all `oc oadp nonadmin` commands, even though non-admin operations do not support namespace selection because of multi-tenancy boundaries. As a consequence, if you specified the namespace flag during a non-admin backup or restore operation, the CLI ignored the provided value and used the active `oc project` context instead, potentially creating resources in an unintended namespace.
```

+

> With this release, the `--namespace` flag is hidden from all `oc oadp nonadmin` subcommands so it no longer is displayed in help output or accepts user input for non-admin operations. As a result, non-admin CLI commands correctly operate only within you current project context, eliminating the risk of namespace mismatch.

Reviewer 2: ```suggestion
With this update, the `--namespace` flag is hidden from all `oc oadp nonadmin` subcommands so that it is no longer displayed in the help output and does not accept user input for non-admin operations. As a result, non-admin CLI commands operate correctly only within your current project context, which eliminates the risk of namespace mismatch.
```

+
link:https://redhat.atlassian.net/browse/OADP-7955[OADP-7955]

Wildcard handling for `excludedClusterScopedResources` no longer causes validation failures::

> Before this update, the {oadp-short} Operator incorrectly appended default resources to `excludedClusterScopedResources` in the `NonAdminBackup` custom resource when a wildcard (`*`) was present. This caused `FailedValidation` errors when excluding all cluster-scoped resources. With this release, the Operator correctly handles the wildcard without appending default resources.

Reviewer 2: ```suggestion
Before this update, the {oadp-short} Operator incorrectly appended default resources to `excludedClusterScopedResources` in the `NonAdminBackup` custom resource (CR) when a wildcard (`*`) was present. As a consequence, `FailedValidation` errors occurred when you excluded all cluster-scoped resources. With this update, the Operator correctly handles the wildcard without appending default resources. As a result, validation failures no longer occur when you use a wildcard to exclude cluster-scoped resources.
```

+
link:https://redhat.atlassian.net/browse/OADP-7897[OADP-7897]

`VolumePolicy` support for phase conditions to allow skipping PVCs::

> Before this update, {oadp-short} volume policies lacked the capability to evaluate and filter `PersistentVolumeClaims` (PVCs) based on their lifecycle status phase. As a consequence, backups could trigger unwanted warnings or errors for unbound or pending PVCs that you excluded, such as in scaled-down serverless environments.

Reviewer 2: ```suggestion
Before this update, {oadp-short} volume policies lacked the capability to evaluate and filter persistent volume claims (PVCs) based on their lifecycle status phase. As a consequence, backups could trigger unwanted warnings or errors for unbound or pending PVCs that you excluded, such as in scaled-down serverless environments.
```

+

> With this release, volume policy actions was improved to natively support the phase condition of PVCs. As a result, you can define custom volume policies in your `ConfigMap` custom resource to automatically skip pending or unbound PVCs during backup operations. For example:

Reviewer 2: ```suggestion
With this update, volume policy actions have been improved to natively support the phase condition of PVCs. As a result, you can define custom volume policies in your `ConfigMap` object to automatically skip pending or unbound PVCs during backup operations. For example:
```

+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: volume-policy
  namespace: velero
data:
  policy.yaml: |
    version: v1
    volumePolicies:
    - conditions:
        pvcPhase: ["Pending", "Lost"]
      action:
        type: skip
----
+
link:https://redhat.atlassian.net/browse/OADP-6428[OADP-6428]

[id="known-issues-1-6-1_{context}"]
== Known issues

`CloudStorage` bucket creation fails on {GCP} with WIF authentication::

> The `CloudStorage` resource fails to detect that a {GCP} bucket does not exist and returns a `BucketCheckError` condition, preventing automatic bucket creation.

Reviewer 2: ```suggestion
When you configure `CloudStorage` on OpenShift clusters on {GCP} by using Workload Identity Federation (WIF) authentication, the `CloudStorage` resource fails to detect that a {GCP} bucket does not exist. As a consequence, the resource returns a `BucketCheckError` condition and prevents automatic bucket creation.
```

+
To work around this problem, use an existing {GCP} bucket when configuring `CloudStorage` on OpenShift clusters on {GCP} using Workload Identity Federation (WIF) authentication. As a result, the controller skips bucket creation and proceeds with the existing bucket for backup storage.
+
link:https://redhat.atlassian.net/browse/OADP-8085[OADP-8085]

Non-admin backups of {odf-short} encrypted volumes fail with {oadp-short}::

> When {oadp-short} Data Mover backs up an encrypted {odf-first} volume, it creates a temporary persistent volume claim (PVC) in the `openshift-adp` namespace. The ceph-csi driver then attempts to fetch the `ceph-csi-kms-token` secret from `openshift-adp` instead of the original application namespace that has the secret. As a consequence, the `NonAdminBackup` fails with an error similar to the following:

Reviewer 2: ```suggestion
When the {oadp-short} Data Mover backs up an encrypted {odf-first} volume, it creates a temporary persistent volume claim (PVC) in the `openshift-adp` namespace. The ceph-csi driver then attempts to fetch the `ceph-csi-kms-token` secret from the `openshift-adp` namespace instead of the original application namespace that contains the secret. As a consequence, the `NonAdminBackup` custom resource (CR) fails with an error similar to the following output:
```

----
+
To work around this problem, copy `ceph-csi-kms-token` and the associated Key Management Service (KMS) config map from the application namespace to `openshift-adp` before running the backup. As a result, the ceph-csi driver can locate the encryption credentials and the backup completes successfully.
+
link:https://redhat.atlassian.net/browse/OADP-7972[OADP-7972]

The Velero server might be terminated with an `OOMKilled` event when accessing large Kopia repositories::
When the Velero server accesses large Kopia repositories, for example, during backup deletion, the Velero server might be terminated by the Out Of Memory (OOM) process. This is typically due to high memory usage loading many Kopia index blobs and related repository metadata.
+
--
To work around this problem, perform the following actions:

> . Check Velero logs around the `OOMKilled` event for repository-open or index-loading activity, and note the approximate maximum memory usage observed for the Velero pod (from oc or kubectl top pods or similar).

Reviewer 2: @weshayutin - this is already included in the 1.6.0 release notes - keep it there or keep it in 1.6.1 ... or for Cthulhu's sake, keep both you deranged madman?

We have the PR ready to remove from 1.6.0

Reviewer 3: oooh.. I want to be a MADMAN!  If it's already in the 1.6.0 RN.. leave it there. 
<img width="1280" height="720" alt="maxresdefault" src="https://github.com/user-attachments/assets/ff73e4ea-dc75-4cd8-9378-78f2e88ab78f" />

Author: @weshayutin, if we remove this Known Issue from 1.6.1 and leave it be in 1.6.0, would that be enough/correct?

Author: Clarified in chat, removed from this release
